src/hooks/useActions.js

```diff
@@ -296,30 +328,103 @@ export default function useActions() {
     [agreementContract]
   )
 
+  // Hoked token manager actions
+  const approveWrappableTokenAmount = useCallback(
+    (amount, onDone = noop) => {
+      if (!wrappableTokenContract || !hookedTokenManagerApp) {
+        return
+      }
+
+      const intent = approveTokenAmount(
+        amount,
+        wrappableTokenContract,
+        hookedTokenManagerApp.address,
+        `Approve ${wrappableToken.symbol}`
+      )
+
+      if (mounted()) {
+        onDone(intent)
+      }
+    },
+    [
+      approveTokenAmount,
+      hookedTokenManagerApp,
+      mounted,
+      wrappableToken,
+      wrappableTokenContract,
+    ]
+  )
+
+  const getHookedTokenManagerAllowance = useCallback(async () => {
+    if (!hookedTokenManagerApp || !wrappableTokenContract) {
+      return
+    }
+    return await getAllowance(
+      wrappableTokenContract,
+      hookedTokenManagerApp.address
+    )
+  }, [hookedTokenManagerApp, getAllowance, wrappableTokenContract])
+
+  const wrap = useCallback(
+    async ({ amount }, onDone = noop) => {
+      let intent = await hookedTokenManagerApp.intent('wrap', [amount], {
+        actAs: account,
+      })
+
+      intent = imposeGasLimit(intent, 1000000)
```

Comment:
Let's use a constant for the gas limit
```suggestion
      intent = imposeGasLimit(intent, 1000000)
```